packages/nimble-components/src/anchor/specs/README.md

```diff
@@ -0,0 +1,111 @@
+# Nimble Anchor
+
+## Overview
+
+The Nimble Anchor is a component used to navigate to a web resource, similar to the HTML anchor element.
+
+### Background
+
+[Visual Design spec](https://xd.adobe.com/view/33ffad4a-eb2c-4241-b8c5-ebfff1faf6f6-66ac/screen/bfadf499-caf5-4ca0-9814-e777fbae0d46)<br>
+[Nimble work item](https://github.com/ni/nimble/issues/324)
+
+---
+
+## Design
+
+### API
+
+[FAST's API documentation](https://github.com/microsoft/fast/blob/e576aa70c22780fffba03097277e2db9a2ec1cd8/packages/web-components/fast-foundation/src/anchor/README.md)
+
+-   _Component Name_: `nimble-anchor`
+-   _Properties/Attributes_: We will have the following properties/attributes in addition to the ones provided by the FAST anchor:
+    -   `appearance`:
+        -   `"text"`: (Default) the "standalone" version in the design doc. Looks like plain text, but gets and underline on hover.
+        -   `"inline-text"`: like `"text"`, but always shows an underline.
+        -   `"outline"`: same as button design
+        -   `"ghost"`: same as button design
+        -   `"block"`: same as button design
+    -   `appearance-variant`:
+        -   `"default"`: `undefined` (as per our common attribute guidelines)
+        -   `"primary"`: applies only to button-based styles and has the same effect as on buttons
+        -   `"prominent"`: applies only to `"text"` and `"inline-text"` appearances. It is the "loud" version from the design doc. Colors the link text green. The name of this attribute value comes from the Breadcrumb.
+    -   `content-hidden`: when set, hides the label and end slot
+    -   `disabled`: when set, makes the anchor inoperable and changes the styling
+-   _Methods_: Unchanged (none)
+-   _Events_: Unchanged (none)
+-   _CSS Classes and Custom Properties that affect the component_: Unchanged (none)
+-   _Slots_: Unchanged (start and end slots)
+
+We are not implementing navigation tabs as part of this effort. That might be implemented in the future by populating a `nimble-tabs` control with `nimble-anchor`s. It may require a new appearance mode.
```

Comment:
I think we need to expand this discussion point a bit further, we have talked about wanting a hyperlink in the following cases:

- hyperlink tab: https://github.com/ni/nimble/issues/479
  - the behavior of cycling would change. Currently roving across tabs immediately activates the tabs. Roving across a link tab should not immediately navigate the page. Should tabs support a mix of both normal tabs and link tabs?
- hyperlink tree item: https://github.com/ni/nimble/issues/562
  - See issue, a dedicated hyperlink tree item should have better keyboard nav behavior
- hyperlink menu item: https://github.com/ni/nimble/issues/324#issuecomment-1201468882
  - Similar to hyperlink tree item, how should links behave in a menu
  - Example from SLS:
      ![image](https://user-images.githubusercontent.com/1588923/203153524-9d64f6ec-6c66-40e5-af95-797241a178f1.png)
- hyperlink button: https://github.com/ni/nimble/issues/324#issue-1120831391

I'm a bit concerned about shoving all of these as different modes into a single component. They likely would have different roles and potentially even different APIs / additional properties needed for the use case.

While the initial implementation does not need to implement every use-case, I think we need to choose a design that can grow into these use-case while being maintainable.

Some potential options, and maybe we need more brainstorming:
1. Design a `nimble-hyperlink` that can support all the known hyperlink use cases discussed above
2. Add hyperlink modes to the above controls that correspond to the use-cases described, ie add href, target, etc attributes to `nimble-tab`, `nimble-menu-item`, etc
3. Split each of these into separate controls, i.e. `nimble-hyperlink`, `nimble-hyperlink-button`, `nimble-hyperlink-tab-item`, `nimble-hyperlink-menu-item`
  - Does this become a large maintenance burden? Would we share styles / templates via patterns?

Reply:
I've added a lot more discussion of the issues. The conclusion I came to was that we should go with a variation of #3. We should create separate components for the hyperlink flavors of tab item, tree item, and menu item, but we can still have the "plain" hyperlink handle the button use case.

Reply:
I see this statement in the discussion:

> then for consistency, we should probably create a separate one for the tabs use case as well

I agree with the point about consistency. But at the same time:

> but we can still have the "plain" hyperlink handle the button use case

Why should buttons be the exception and be bundled into the hyperlink use case?

Reply:
From offline discussion about aria roles for links, an example of [menu button links](https://www.w3.org/WAI/ARIA/apg/example-index/menu-button/menu-button-links) from aria which links to a relevant discussion on [Roles That Automatically Hide Semantics by Making Their Descendants Presentational](https://www.w3.org/WAI/ARIA/apg/practices/hiding-semantics/#children-presentational).

Reply:
Brainstormed with @atmgrifter00 a bit and additional thoughts to consider:

- Having separate components for button kind of aligns with our other button variations:
    ```html
    <nimble-button>
    <nimble-menu-button>
    <nimble-toggle-button>
    <nimble-anchor-button>
    ```
- API-wise is there a benefit to being able to switch the visual of a link with just attribute changes? It does not seem like a useful degree of freedom. The API is also making it very easy to configure things incorrectly and end up with a very different visual (lots of invalid permutations of appearance and appearance-variant).
- For consistency we would either have a single anchor that supports all the different visual modes are separate anchors. If we extrapolate to having many visual modes for anchors then, similar to icons, if they are all bundled into one component you pay the complexity and size cost of all variations even if only using a few.

Some additional thoughts:
- The visuals for buttons and anchors are very different. Buttons are widgets that have a height and width (always a box) while anchors are runs of text that span and can wrap multiple lines. Seems strange for the same control to have such different layout and sizing behaviors.
- Separate components let us have different Angular directive selectors based on the element name in case we need additional degrees of freedom for directive binding.

Reply:
I've reworked the spec to split the anchor and anchor button into two separate components. I've listed some of your points as arguments for the split. Really there were no good arguments I could come up with for keeping them as one component. Hopefully it addresses this thread.